### DISCUSSION TITLE
Combine Different Images to One, part 2 (and GUI)

### DISCUSSION THREAD

**Discussion by Jay123210599** [ImageMagick/ImageMagick#7361] [Help]
### ImageMagick version

7.1.1-27

### Operating system, version and so on

Windows 11

### Description

Is there an official GUI for ImageMagick so we can perform our tasks easier? If so, how do we use it?

Also, how do I combine images of different resolutions by making the larger file smaller, and/or make both images equal in length, height, and width?

**Comment by snibgo** [comment]
> Is there an official GUI for ImageMagick ...

No.

> how do I combine images of different resolutions by making the larger file smaller ...

How do you want to make it "smaller"? You might crop the larger image to be the same size as the smaller, or you might resize it, which might change the aspect ratio. Or something else, such as a crop and a resize.

What do you want to do if one image has a greater width than the other, but a smaller height?

**Reply by Jay123210599** [reply]
resize, not crop. Combine them, of course.

**Comment by snibgo** [comment]
I don't know what you mean by "combine". I assume you want to put them side-by-side in a single image.
```
magick inA.png inB.png -set option:WW %[fx:min(u.w,v.w)] -set option:HH %[fx:min(u.h,v.h)] -resize %[WW]x%[HH]! +append +repage out.png
```
This resizes both images so the width of each is the minimum of the widths of the two inputs, and similarly the height.

Then it puts them side-by-side, and writes the result to a file.

If used in a Windows BAT script, double each percent: `%%`.

**Comment by fmw42** [comment]
What do you mean by "combine"?  Put them side by side, or overlay one on the other, or make 3 single channel grayscale images into one 3 channel color image? 

What do you want to do if one image is large in one dimension and smaller in the other?  If you want to resize, then to make them the same size, after resizing, one would have to crop some off the larger dimension.  Alternately, one could resize to the other dimension and pad the opposite dimension.  Alternately, resizing both dimensions separately might cause visible distortion.
